File: sorting/python/selection_sort.py
#!/usr/bin/env python3


def selection_sort(items: list[int]) -> list[int]:
    for i in range(len(items) - 1):
        minIndex = i

        for j in range(i + 1, len(items)):
            if items[j] < items[minIndex]:
                minIndex = j

        if minIndex != i:
            items[i], items[minIndex] = items[minIndex], items[i]

    return items

print(selection_sort([-9, -100, 9, 0, 100, 4]))

File: sorting/rust/sorting/src/selection_sort.rs
pub fn selection_sort(arr: &mut [i32]) {
    for i in 0..arr.len() - 1 {
        let mut minIndex = i;

        for j in i + 1..arr.len() {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }

        if (minIndex != i) {
            arr.swap(i, minIndex);
        }
    }
}
